libraries/AP_RangeFinder/AP_RangeFinder_RDS02UF.cpp

```diff
@@ -0,0 +1,175 @@
+        case RDS02UF_PARSE_STATE::STATE7_LENGTH_H:{
+            // length_high
+            uint8_t read_len = data << 8 | parser_buffer[parser_index-1];
+            // rds02uf data length is always 10
+            if (read_len == RDS02UF_DATA_LEN) {
+                parser_buffer[parser_index++] = data;
+                decode_state = RDS02UF_PARSE_STATE::STATE8_REAL_DATA;
+            } else {
+                parser_index = 0;
+                decode_state = RDS02UF_PARSE_STATE::STATE0_SYNC_1;
+            }
+            break;
+        }
+        case RDS02UF_PARSE_STATE::STATE8_REAL_DATA: // real_data
+            parser_buffer[parser_index++] = data;
+            if (parser_index == (RDS02UF_HEAD_LEN + RDS02UF_PRE_DATA_LEN + RDS02UF_DATA_LEN) ) {
+                decode_state = RDS02UF_PARSE_STATE::STATE9_CRC;
+            }
+            break;
+        case RDS02UF_PARSE_STATE::STATE9_CRC:{
+            // crc
+	        const uint8_t crc_data = crc8_rds02uf(&parser_buffer[2], RDS02UF_PRE_DATA_LEN + RDS02UF_DATA_LEN);
+            if (data == crc_data || data == RDS02UF_IGNORE_CRC) {
+                // 0xff indicates that the current frame does not need to be checked.
+                parser_buffer[parser_index++] = data;
+                decode_state = RDS02UF_PARSE_STATE::STATE10_END_1;
+            } else {
+                parser_index = 0;
+                decode_state = RDS02UF_PARSE_STATE::STATE0_SYNC_1;
+            }
+            break;
+        }
+        case RDS02UF_PARSE_STATE::STATE10_END_1:
+            if (data == RDS02_END) {
+                parser_buffer[parser_index++] = data;
+                decode_state = RDS02UF_PARSE_STATE::STATE11_END_2;
+            } else {
+                parser_index = 0;
+                decode_state = RDS02UF_PARSE_STATE::STATE0_SYNC_1;
+            }
+            break;
+        case RDS02UF_PARSE_STATE::STATE11_END_2:{
+            const uint16_t fc_code = (parser_buffer[RDS02UF_PRODUCTS_FC_INDEX_H] << 8 | parser_buffer[RDS02UF_PRODUCTS_FC_INDEX_L]);
+            const uint8_t err_code = parser_buffer[RDS02UF_ERROR_CODE_INDEX];
+            if (data == RDS02_END) {
+                if (fc_code == RDS02UF_UAV_PRODUCTS_ID && err_code == RDS02UF_NO_ERR) {
+                    // get target information
+                    const uint16_t read_info_fc = (parser_buffer[RDS02_TARGET_FC_INDEX_H] << 8 | parser_buffer[RDS02_TARGET_FC_INDEX_L]);
+                    if ((read_info_fc & RDS02UF_IGNORE_ID_BYTE) == RDS02_TARGET_INFO_FC) {
+                        // read_info_fc = 0x70C + ID * 0x10, ID: 0~0xF
+                        distance_m = (parser_buffer[RDS02_DATA_Y_INDEX_H] * 256 + parser_buffer[RDS02_DATA_Y_INDEX_L]) * 0.01f;
+                        ret = true;
+                        state.last_reading_ms = AP_HAL::millis();
+                    }
+                }
+            }
+			parser_index = 0;
+			decode_state = RDS02UF_PARSE_STATE::STATE0_SYNC_1;
+			break;
+        }
```

Comment:
check index == sizeof(buffer), and reset at this point, so we don't put any more bytes in